### DISCUSSION TITLE
How to configure postcss autoprefixer  and stitches in a NEXTJS project?

### DISCUSSION THREAD

**Discussion by LuisFerreirinhaKK** [stitchesjs/stitches#1024] [Help] [answered]
I've finished a Nextjs project with stitches and modern css features like "gap" ( @https://caniuse.com/?search=gap) and I ran into a few problems on old safari browsers (<14). 

Since Next.js allow us to configure the target browsers (for **Autoprefixer** and compiled css features) through **Browserslist** I tried to configure browserlist as bellow but nothing happened, I'm still facing the same problems:

`"browserslist": [
    ">= 0.5%",
    "chrome >= 42",
    "and_chr >= 42",
    "and_ff >= 38",
    "android >= 4.4",
    "edge >= 12",
    "firefox >= 38",
    "ie >= 11",
    "ios_saf >= 9",
    "safari >= 9",
    "node 10"
  ]`.

Does Stitches requires some other tool to support older browsers?
Thanks in advance

**Comment by predaytor** [comment, marked as answer]
You can configure postcss and its features such as autoprefixer etc. But keep in mind that Stitches is currently only a runtime, which means postcss can only handle styles from **already mounted** components (conditional rendering is not supported) and only at **build time**. In dev mode - on initial SSR request (hot reload will reset styles without postcss processing).
Things like flex gap polyfill are difficult to implement (for example, it requires a special additional layout).

`_document.tsx`:
```tsx
import React from 'react';
import postCss, { AcceptedPlugin } from 'postcss';
import postCssPresetEnv from 'postcss-preset-env';
import NextDocument, { DocumentContext, Html, Head, Main, NextScript } from 'next/document';
import { getCssText } from 'stitches.config';

export default class Document extends NextDocument {
	/* styles from already mounted components are currently being processed at build time only */
	static async getInitialProps(ctx: DocumentContext) {
		const cssText = getCssText();
		const { css } = await postCss([
			postCssPresetEnv({
				browsers: 'last 2 versions',
				stage: 2,
				autoprefixer: {},
			}) as AcceptedPlugin,
		]).process(cssText, {
			from: undefined,
		});

		const { styles, ...initialProps } = await NextDocument.getInitialProps(ctx);

		return {
			...initialProps,
			styles: [
				<React.Fragment key="styles">
					{styles}
					<style id="stitches" dangerouslySetInnerHTML={{ __html: css }} />
				</React.Fragment>,
			],
		};
	}

	render() {
		return (
			<Html lang="en">
				<Head />

				<body>
					<Main />
					<NextScript />
				</body>
			</Html>
		);
	}
}

```
Soon (I hope) will be available api for *static extraction*, which will solve problem.

**Reply by LuisFerreirinhaKK** [reply]
Thank you very much for your feedback and brief and objective explanation! Really looking forward for that api for static extraction!

**Comment by mattp0123** [comment]
I think maybe this should be write into the [docs](https://stitches.dev/blog/using-nextjs-with-stitches)? But I don't know who should I notify.
You see there is a [same question I asked](https://github.com/stitchesjs/stitches/discussions/983) two months ago 😂
